catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -2482,73 +2418,45 @@ protected ResourceResponse getResource(ResourceRequest resourceRequest, boolean
                 resourceSourceName = resolvedSourceId;
             }
 
-            String key;
-            try {
-                key = new CacheKey(metacard, resourceRequest).generateKey();
```

Comment:
It may still exist, but should only exist behind the `ResourceCache` class and shouldn't be expost to clients anymore. And if we keep a class, I hope we won't have something that creates a `String` and instead simply pass around a `CacheKey` object that has proper `hashCode()` and `equals()` methods.